walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> overlapEvents = new ArrayList<Event>();
```

Comment:
Looks like your algorithm is a series of transformations from a List to another List and so forth.
- Consider documenting how your algorithm works. You could write some high-level functions, or break down this function into helper functions that implement steps of the algorithm.
- If the function remains one piece, I would move these List initializations to the step they correspond to. No need for the reader to wonder what this for until they get there.